src/storage/mvcc/reader/scanner/util.rs

```diff
@@ -81,7 +83,29 @@ where
     assert!(write.short_value.is_none());
     let seek_key = user_key.clone().append_ts(write.start_ts);
     default_cursor.near_seek(&seek_key, &mut statistics.data)?;
-    assert!(default_cursor.valid());
+    if !default_cursor.valid() {
+        CRITICAL_ERROR
+            .with_label_values(&["default value not found"])
+            .inc();
+        if panic_when_unexpected_key_or_data() {
+            set_panic_mark();
+            panic!(
+                "default value not found for key {}, write: {:?}",
+                hex::encode_upper(&user_key.to_raw()?),
+                write
+            );
+        } else {
+            error!(
+                "default value not found";
+                "key" => log_wrappers::Key(&user_key.to_raw()?),
+                "write" => ?write,
+            );
+            return Err(Error::DefaultNotFound {
+                key: user_key.to_raw()?,
+                write: write.clone(),
+            });
```

Comment:
Seems L90-L106 appears in several places, better to extract into a function.

Reply:
That will cover up the spot where the error occurs so that we won't know either `scan` or `point_get` cause this from error log.  Are you sure it is okay?

Reply:
You can pass a parameter to indicate where it is called.